# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/chromium/cr145-rustc_nightly_capability.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

From ba37d384c411b7db1da453593f68b9e9bf249534 Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Mon, 19 Jan 2026 23:07:16 +0100
Subject: [PATCH] Assume rustc is nightly if building Chromium

Nightly is required to build full Chromium.
This variable was introduced to build just V8 with ancient Rust,
and is only used to bring incompatibilities with current Rust versions.
---
 build/config/rust.gni | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/config/rust.gni b/build/config/rust.gni
index 36a1fdd350f4a..f88d57f45996c 100644
--- a/build/config/rust.gni
+++ b/build/config/rust.gni
@@ -121,7 +121,7 @@ use_chromium_rust_toolchain = rust_sysroot_absolute == ""
 # `rustc_nightly_capability = false` will make the build avoid the use of
 # Rust nightly features. There are no bots that test this and there is no
 # guarantee it will work, but we accept patches for this configuration.
-rustc_nightly_capability = use_chromium_rust_toolchain
+rustc_nightly_capability = use_chromium_rust_toolchain || build_with_chromium
 
 # Platform support for the Rust toolchain.
 chromium_toolchain_supports_platform = !is_wasm
